apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/entity/blockchain/BlockImpl.java

```
@@ -228,12 +209,21 @@ public long getPreviousBlockId() {
 
     @Override
     public byte[] getGeneratorPublicKey() {
-        if (generatorPublicKey == null) {
-            generatorPublicKey = lookupAccountService().getPublicKeyByteArray(generatorId);
-        }
+//        if (generatorPublicKey == null) {
```

need to throw here exception if generatorKey was not set

---

Added throw exception

---

Reverted code back, cause we want : 
check if that field is empty and assign new PubKey value using 'block.getGeneratorId()' method's value.